packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -0,0 +1,3369 @@
+import { getEtherscanApiHost } from './utils/etherscan';
+import * as etherscanUtils from './utils/etherscan';
+
+const ACCOUNT_MOCK = '0x6bf137f335ea1b8f193b8f6ea92561a60d23a207';
+const ACCOUNT_2_MOCK = '0x08f137f335ea1b8f193b8f6ea92561a60d23a211';
+const ACCOUNT_3_MOCK = '0xe688b84b23f322a994a53dbf8e15fa82cdb71127';
+const infuraProjectId = '341eacb578dd44a1a049cbc5f6fd4035';
+
+const networkClientConfiguration = {
+  type: NetworkClientType.Infura,
+  network: NetworkType.goerli,
+  chainId: BUILT_IN_NETWORKS[NetworkType.goerli].chainId,
+  infuraProjectId,
+  ticker: BUILT_IN_NETWORKS[NetworkType.goerli].ticker,
+} as const;
+
+const sepoliaNetworkClientConfiguration = {
+  type: NetworkClientType.Infura,
+  network: NetworkType.sepolia,
+  chainId: BUILT_IN_NETWORKS[NetworkType.sepolia].chainId,
+  infuraProjectId,
+  ticker: BUILT_IN_NETWORKS[NetworkType.sepolia].ticker,
+} as const;
+
+const mainnetNetworkClientConfiguration = {
+  type: NetworkClientType.Infura,
+  network: NetworkType.mainnet,
+  chainId: BUILT_IN_NETWORKS[NetworkType.mainnet].chainId,
+  infuraProjectId,
+  ticker: BUILT_IN_NETWORKS[NetworkType.mainnet].ticker,
+} as const;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const newController = async (options: any = {}) => {
+  const messenger = new ControllerMessenger();
+  const networkController = new NetworkController({
+    messenger: messenger.getRestricted({ name: 'NetworkController' }),
+    trackMetaMetricsEvent: () => {
+      // noop
+    },
+    infuraProjectId,
+  });
+  await networkController.initializeProvider();
+  const { provider, blockTracker } =
+    networkController.getProviderAndBlockTracker();
+
+  const approvalController = new ApprovalController({
+    messenger: messenger.getRestricted({
+      name: 'ApprovalController',
+    }),
+    showApprovalRequest: jest.fn(),
+    typesExcludedFromRateLimiting: [ApprovalType.Transaction],
+  });
+
+  const { state, config, ...opts } = options;
+
+  const transactionController = new TransactionController(
+    {
+      provider,
+      blockTracker,
+      messenger,
+      onNetworkStateChange: () => {
+        // noop
+      },
+      getCurrentNetworkEIP1559Compatibility:
+        networkController.getEIP1559Compatibility.bind(networkController),
+      getNetworkClientRegistry:
+        networkController.getNetworkClientRegistry.bind(networkController),
+      findNetworkClientIdByChainId:
+        networkController.findNetworkClientIdByChainId.bind(networkController),
+      getNetworkClientById:
+        networkController.getNetworkClientById.bind(networkController),
+      getNetworkState: () => networkController.state,
+      getSelectedAddress: () => '0xdeadbeef',
+      getPermittedAccounts: () => [ACCOUNT_MOCK],
+      ...opts,
+    },
+    {
+      // TODO(JL): fix this type
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      sign: async (transaction: any) => transaction,
```

Comment:
fix this

Reply:
working on this

Reply:
https://github.com/MetaMask/core/pull/3643/commits/07fb74ca115bbf7835c4c894a964eae45cb87c0f